### PR TITLE
feat: migrate repo from cairo 2.2.0 - 2.6.4

### PR DESCRIPTION
**Pull request by EjembiEmmanuel**

### CODE REVIEW

File: src/components/erc2981.cairo [hunk 1]
```diff
@@ -21,16 +28,25 @@ mod ERC2981 {
         ERC2981_token_fee_denominator: LegacyMap<u256, u256>,
     }
 
-    #[external(v0)]
-    impl ERC2981Impl of IERC2981<ContractState> {
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {}
```
Comment (tekkac) [inline src/components/erc2981.cairo]: Not sure this is needed

**Review by tekkac — DISMISSED** [review]
lgtm :muscle: 
minor comments otw good to go
good job!

File: src/components/erc2981.cairo (hunk 1 above)
Comment (EjembiEmmanuel) [reply src/components/erc2981.cairo]: Alright, will remove it.

**Comment by EjembiEmmanuel** [comment]
@tekkac done!

File: src/components/erc2981.cairo (hunk 1 above)
Comment (EjembiEmmanuel) [reply src/components/erc2981.cairo]: @tekkac done!

**Review by tekkac — APPROVED** [review]
:ok_hand:

**Pull request merged**